File: backend/starry-accessory/src/main/java/com/amy/starryaccessory/dto/Purchase.java
package com.amy.starryaccessory.dto;

import com.amy.starryaccessory.entity.Address;
import com.amy.starryaccessory.entity.Customer;
import com.amy.starryaccessory.entity.Order;
import com.amy.starryaccessory.entity.OrderItem;

import lombok.Data;
import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;

}

File: frontend/starry-accessory/src/models/Purchase.ts
import { Address } from "./Address";
import { Customer } from "./Customer";
import { Order } from "./Order";
import { OrderItem } from "./OrderItem";

export class Purchase {
  customer!: Customer;
  shippingAddress!: Address;
  billingAddress!: Address;
  order!: Order;
  orderItems!: OrderItem[];
}
